sphinx/ext/autosummary/__init__.py

```diff
@@ -234,9 +234,12 @@ def run(self) -> list[Node]:
         nodes = self.get_table(items)
 
         if 'toctree' in self.options:
-            dirname = posixpath.dirname(self.env.docname)
-
             tree_prefix = self.options['toctree'].strip()
+            if posixpath.isabs(tree_prefix):
```

Comment:
Maybe consider using SEP rather than '/'. I'd take inpiration from how the ``:doc:`` role is implemented
```suggestion
            if tree_prefix.startswith('/'):
```

Reply:
Done in cdca8ed26.

I didn't use `os.sep` because it would cause a little bit of inconsistency with the rest of Autosummary on Windows, which uses `posixpath` for manipulations regardless of root platform, and `os.sep` on Windows is `\`.  There's `os.altsep` that's `/` on Windows, but I didn't immediately want to go that root, because I was a bit concerned that there'd be knock-on effects from the rest of the module still using `posixpath` and thought that changing the path handling more completely might want to be a different PR if there was a reason to do it.

Reply:
Sorry; SEP is a constant that we define -- see ``osutil``.

A

Reply:
Ah sorry, I misunderstood.  That seems much preferable - the magic `"/"` constant didn't feel great. Changed in 879b6b259.

Reply:
(Aside: nice to see a fellow Leodensian!)

By way of an apology, autosummary has a lot of accumulated cruft and I tend to avoid looking at it out of habit. I'll try and look at this soon, but do feel free to ping. I'm currently working on the Python 3.12 documentation ahead of release next week, so as hopefully will get to this no later than mid-October.

Thanks,
Adam

Reply:
No worries at all, thanks - I'm in no huge rush.  If the only thing holding this up now is just your workload, I can just put it aside for a month and give it a bump if needed then.

I'm actually separately pushing to reduce our codebase's dependence on `autosummary` for similar reasons, and because (imo) it can be easy to fall into not-so-great documentation-structure habits if it's overused, which we certainly do.

(Yeah, it's not so common to see northerners around!  I grew up in York, went down south for undergrad and PhD - looks like you're similar - and now that's all done so I'm back.)